src/components/ProjectsList/ProjectsList.jsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import styles from './ProjectsList.module.css';
+import { PROJECTS } from '../../constants/constants';
+import { setQuery, updateFilteredItems } from '../../actions';
+import { filterProjects } from '../../utils/filterProjects';
+
+const DEBOUNCE_DELAY = 300;
+
+export const ProjectsList = () => {
+    const dispatch = useDispatch();
+    const query = useSelector((state) => state.projects.query);
+    const filteredItems = useSelector((state) => state.projects.filteredItems);
+    const isEmpty = filteredItems.length === 0;
+
+    useEffect(() => {
+        const handler = setTimeout(() => {
+            const results = filterProjects(PROJECTS, query);
```

Comment:
Right now this component is responsible for handling user input, filtering the cards and displaying them. These are 3 different things. Think about moving each of these responsibilities into their own module.